### PR TITLE
refactor(restclient): do not require input argument on fully optional interface

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #13** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/13" target="_blank">Graphite</a> 
    * **PR #14** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/14" target="_blank">Graphite</a> 
      * **PR #15** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/15" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/15?utm_source=stack-comment).

File: src/lib/primitives/restclient/restclient.ts [hunk 1]
```diff
@@ -514,8 +514,16 @@ export async function $restclient(definition: OpenapiV3, options: Partial<Restcl
                         headerParameters?.map((p) => `${objectProperty(p.name)}${p.required === true ? '' : '?'}: ${p.type}`) ??
                         []
                     ).join(', ')
-                    const headerOptionalStr = headerParameters?.every((q) => q.required === true) ? '' : '?'
 
+                    const hasRequiredBody = request?.declaration !== undefined
+                    const hasRequiredPathArgument = pathArguments.length > 0
+                    const hasRequiredQueryArguments = queryArguments.length > 0
```
Comment (MickVanDuijn) [inline src/lib/primitives/restclient/restclient.ts:520]:
```suggestion
                    const hasRequiredQueryArguments = queryArguments?.some((q) => q.required === true)
```

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/15?utm_source=stack-comment).


Job Reference: 3k1UzKXl5ZWLtCzZn7QI

**Pull request merged**